lib/matplotlib/axes/_axes.py

```diff
@@ -4364,7 +4367,14 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xsize,
             if facecolors is None:
                 facecolors = kwcolor
 
-        if edgecolors is None and not mpl.rcParams['_internal.classic_mode']:
+        edge_from_c = False
+        if edgecolors is None and c is not None:
+            edge_from_c = True
+
+        facecolors_none = type(facecolors) == str and facecolors == 'none'
```

Comment:
```suggestion
        facecolors_none = isinstance(facecolors, str) and facecolors == 'none'
```

We prefer `isinstance` over `==` on types.  I am guessing that this check is here to guard against the "truth value of an array is ambiguous" error and while I hope no one is feeding us `str` subclasses, we should be forgiving if they do (as we do not actually about the string-y ness of this input _other_ than avoiding a numpy complication).

Reply:
Updated